### PR TITLE
C360106 Verify the Service point in the Item Circulation history (firebird) (TaaS)

### PR DESCRIPTION
**Pull request by ernestyedigaryan1**
[C360106](https://foliotest.testrail.io/index.php?/cases/view/360106)

[Allure Report](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/4714/allure/)

### CODE REVIEW

File: cypress/support/fragments/inventory/inventorySearchAndFilter.js [hunk 1]
```diff
@@ -653,4 +653,11 @@ export default {
   verifyInstanceDetailsViewAbsent() {
     cy.expect(instanceDetailsSection.absent());
   },
+
+  searchByStatus(status) {
```
Comment (manowarqqq) [inline cypress/support/fragments/inventory/inventorySearchAndFilter.js:657]: method doesn't correspond to action. Need to click checkbox, not to verify that it's exists

File: cypress/e2e/circulation-log/verify-service-point-in-circ-history.cy.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      // #2 Search for an Item with status "Available" by clicking on the "Item status" accordion and selecting "Available" checkbox
+      InventorySearchAndFilter.searchByStatus('Available');
+      // #3 Click on the link with Items barcode to view the Item
+      InventorySearchAndFilter.searchByParameter('Barcode', ITEM_BARCODE);
+      ItemRecordView.waitLoading();
+      // #4 Click on the "Circulation history" accordion to expand it => Check the "Most recent check in"
+      ItemRecordView.checkItemCirculationHistory('-', '-', '-');
+      // #6 Navigate to the "Check in" app and check in that Item by pasting the copied Items barcode to the input filled => Hit "Enter" button
+      cy.visit(TopMenu.checkInPath);
+      CheckInActions.checkInItem(ITEM_BARCODE);
```
Comment (manowarqqq) [inline cypress/e2e/circulation-log/verify-service-point-in-circ-history.cy.js:79]: Need an assertion that Item checked In

File: cypress/e2e/circulation-log/verify-service-point-in-circ-history.cy.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      // #1 Go to the "Inventory" => Select "Item" toggle
+      InventorySearchAndFilter.switchToItem();
+      // #2 Search for an Item with status "Available" by clicking on the "Item status" accordion and selecting "Available" checkbox
+      InventorySearchAndFilter.searchByStatus('Available');
+      // #3 Click on the link with Items barcode to view the Item
+      InventorySearchAndFilter.searchByParameter('Barcode', ITEM_BARCODE);
+      ItemRecordView.waitLoading();
+      // #4 Click on the "Circulation history" accordion to expand it => Check the "Most recent check in"
+      ItemRecordView.checkItemCirculationHistory('-', '-', '-');
+      // #6 Navigate to the "Check in" app and check in that Item by pasting the copied Items barcode to the input filled => Hit "Enter" button
```
[the remark this answers is not in the document]
Comment (ernestyedigaryan1) [reply cypress/e2e/circulation-log/verify-service-point-in-circ-history.cy.js]: I don't think it is required to check the notification in this specific test case
If you think it is mandatory to check, I can add an assertion

[the remark this answers is not in the document]
Comment (manowarqqq) [reply cypress/e2e/circulation-log/verify-service-point-in-circ-history.cy.js]: This is in the test case, I think you should stick to it

**Review by manowarqqq — APPROVED** [review]

File: cypress/e2e/circulation-log/verify-service-point-in-circ-history.cy.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+describe('Circulation log', () => {
+  const testData = {
+    folioInstances: InventoryInstances.generateFolioInstances(),
+    servicePoint: ServicePoints.getDefaultServicePointWithPickUpLocation(),
+    servicePointS1: ServicePoints.getDefaultServicePointWithPickUpLocation(),
+    requestsId: '',
+  };
+  let userData;
+  let ITEM_BARCODE;
```
Comment (nayimovag) [inline cypress/e2e/circulation-log/verify-service-point-in-circ-history.cy.js:23]: no need for uppercase if variable is not constant

File: cypress/e2e/circulation-log/verify-service-point-in-circ-history.cy.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+import CheckInActions from '../../support/fragments/check-in-actions/checkInActions';
+import InventorySearchAndFilter from '../../support/fragments/inventory/inventorySearchAndFilter';
+import ItemRecordView from '../../support/fragments/inventory/item/itemRecordView';
+import ConfirmItemInModal from '../../support/fragments/check-in-actions/confirmItemInModal';
+
+describe('Circulation log', () => {
+  const testData = {
+    folioInstances: InventoryInstances.generateFolioInstances(),
+    servicePoint: ServicePoints.getDefaultServicePointWithPickUpLocation(),
+    servicePointS1: ServicePoints.getDefaultServicePointWithPickUpLocation(),
```
Comment (nayimovag) [inline cypress/e2e/circulation-log/verify-service-point-in-circ-history.cy.js:19]: please choose better variable names, it's unclear what S1 means in servicePointS1

**Review by nayimovag — APPROVED** [review]

**Review by nayimovag — APPROVED** [review]

**Pull request merged**